esiclient/v1/node_port.py

```diff
@@ -0,0 +1,123 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import json
+import logging
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+
+def _get_attrs_node(parsed_args):
+    attrs = {}
+    field_list = ["name", "driver", "driver_info", "resource_class"]
+    for (k, v) in vars(parsed_args).items():
+        if k in field_list and v is not None:
+            attrs[k] = v
+    driver_info_values = attrs.get("driver_info")
+    if driver_info_values:
+        driver_info_dict = param_array_to_dict(driver_info_values)
+        attrs.update({"driver_info": driver_info_dict})
+    return attrs
+
+
+def _get_attrs_port(parsed_args):
+    attrs = {}
+    field_list = ["address", "node_uuid", "local_link_connection",
+                  "physical_network"]
+    for (k, v) in vars(parsed_args).items():
+        if k in field_list and v is not None:
+            attrs[k] = v
+    local_link_connection_values = attrs.get("local_link_connection")
+    if local_link_connection_values:
+        local_link_connection_dict = param_array_to_dict(
+            local_link_connection_values)
+        attrs.update({"local_link_connection": local_link_connection_dict})
+    return attrs
+
+
+def param_array_to_dict(param_array):
```

Comment:
If the values are literals, you could replace this function with:

```
def param_array_to_dict(param_array):
    return dict(x.split('=', 1) for x in param_array)
```

If sometimes values will be a JSON object, you could instead write:

```
def param_array_to_dict(param_array):
    def _maybe_json(v):
        try:
            return json.loads(v)
        except ValueError:
            return v

    return dict((y[0], _maybe_json(y[1])) for y in [
        x.split('=', 1) for x in param_array
    ])
```

Reply:
Thanks! I've changed the code. And I add another try except block to check x.split('=', 1).